docs/specs/captcha.md

```diff
@@ -1,41 +1,328 @@
+  * [Authentication Flow](#authentication-flow)
+    + [Captcha in Authentication Flow configuration](#captcha-in-authentication-flow-configuration)
+    + [Default behavior](#default-behavior)
+    + [Captcha in Authentication Flow API](#captcha-in-authentication-flow-api)
+    + [Advanced use case: Require Captcha at a specific branch, instead of right after flow creation](#advanced-use-case-require-captcha-at-a-specific-branch-instead-of-right-after-flow-creation)
+    + [Future use cases](#future-use-cases)
+      - [Future use case: Use different Captcha providers in different flows](#future-use-case-use-different-captcha-providers-in-different-flows)
+      - [Future use case: Allow fallback in providers](#future-use-case-allow-fallback-in-providers)
+      - [Future use case: Fail-open instead of Fail-close](#future-use-case-fail-open-instead-of-fail-close)
+      - [Future use case: Use risk assessment to determine whether captcha is required](#future-use-case-use-risk-assessment-to-determine-whether-captcha-is-required)
+  * [Audit log](#audit-log)
+  * [Study on Captcha providers](#study-on-captcha-providers)
+    + [Geetest v4](#geetest-v4)
+    + [Geetest v3](#geetest-v3)
+    + [hCaptcha](#hcaptcha)
+    + [reCAPTCHA v2](#recaptcha-v2)
+    + [reCAPTCHA v3](#recaptcha-v3)
+    + [reCAPTCHA Enterprise](#recaptcha-enterprise)
+    + [Cloudflare Turnstile](#cloudflare-turnstile)
+    + [Arkose Labs Bot Manager](#arkose-labs-bot-manager)
+    + [Tencent Captcha](#tencent-captcha)
 
-## Configurations
+## Old configuration
 
-### authgear.yaml
+> The old configuration **IS NOT** used by Authentication Flow.
+> To configure Captcha providers for an Authentication Flow, the new configuration must be used.
+
+See [here](./captcha_legacy.md#configuration) for the documentation of the old configuration.
 
-Before using any workflow with captcha enabled, the captcha provider must be configured in `authgear.yaml`:
+## New Configuration
+
+The section documents the new configuration.
+
+### authgear.yaml
 
 ```yaml
 captcha:
-  provider: cloudflare
+  enabled: true
+  providers:
+  - type: cloudflare
+    alias: cloudflare
 ```
 
-- `provider`
-  - The captcha provider. Currently, only `cloudflare` is supported.
+- `captcha.enabled`: Boolean. If it is true, the new configuration is used.
+- `captcha.providers`: An array of Captcha provider configuration. The actual shape depends on the `type` property.
+- `captcha.providers.type`: Required. The type of the Captcha provider. Valid values are `cloudflare` and `recaptchav2`.
+- `captcha.providers.alias`: Required. The unique identifier to the Captcha provider. It is used in other parts of the configuration to refer to this particular Captcha provider. For example, the project can configured a number of Captcha providers, but only uses one of them in a particular Authentication Flow.
 
-### authgear.secrets.yaml
+Other fields are specific to `type`.
+
+#### `type: cloudflare`
```

Comment:
Seems only cloudfare here, is recaptcha v2 intended to left blank for me to fill in when implementing? 🙏

Reply:
Added a commit to address this.